front/lib/api/data_source_view.ts

```diff
@@ -276,3 +279,51 @@ export async function getContentNodesForDataSourceView(
     total: contentNodesResult.total,
   });
 }
+
+export async function handlePatchDataSourceView(
```

Comment:
Moved from a private endpoint to lib and refactored a bit to return `Result` instead of directly a next response.
Declare it's params type (here: `PatchDataSourceViewType`).

Then:
- in the public api endpoint, there is the public request schema (& response type)
- in the private api endpoint, there is the private request schema (& response type)
- both endpoints call this logic method.

This refactoring pattern has been applied a few times.